x509/pkcs8.go

```diff
@@ -94,9 +102,32 @@ func MarshalPKCS8PrivateKey(key interface{}) ([]byte, error) {
 			return nil, errors.New("x509: failed to marshal EC private key while building PKCS#8: " + err.Error())
 		}
 
+	case ed25519.PrivateKey:
+		privKey.Algo = pkix.AlgorithmIdentifier{Algorithm: OIDPublicKeyEd25519}
+		privKey.PrivateKey = MarshalEd25519PrivateKey(k)
+
 	default:
 		return nil, fmt.Errorf("x509: unknown key type while marshalling PKCS#8: %T", key)
 	}
 
 	return asn1.Marshal(privKey)
 }
+
+// MarshalEd25519PrivateKey converts an Ed25519 private key to ASN.1 DER encoded form
+// (as an OCTET STRING holding the key seed).
+func MarshalEd25519PrivateKey(key ed25519.PrivateKey) []byte {
+	b, _ := asn1.Marshal(key.Seed())
```

Comment:
You're certain that this will never return an error? It certainly doesn't look like it should ever return an error, but I'm wondering whether that can be assumed.

Reply:
Fair point, added `error` return.